src/main/kotlin/org/rust/ide/refactoring/extractFunction/ui.kt

```diff
@@ -47,12 +53,50 @@ interface ExtractFunctionUi {
     fun extract(config: RsExtractFunctionConfig, callback: () -> Unit)
 }
 
+sealed class ExtractTextField(val component: JComponent) {
+    data class SimpleTextField(val field: JBTextField) : ExtractTextField(field)
+    data class SuggestionTextField(val field: NameSuggestionsField) : ExtractTextField(field)
+
+    val name: String
+        get() =
+            when (this) {
+                is SimpleTextField -> this.field.text
+                is SuggestionTextField -> this.field.enteredName
+            }
+
+    fun addChangeListener(listener: () -> Unit) {
+        when (this) {
+            is SimpleTextField -> this.field.document.addDocumentListener(object : DocumentListener {
+                override fun changedUpdate(p0: DocumentEvent?) {
+                    listener()
+                }
+
+                override fun insertUpdate(p0: DocumentEvent?) {
+                    listener()
+                }
+
+                override fun removeUpdate(p0: DocumentEvent?) {
+                    listener()
+                }
+            })
+            is SuggestionTextField -> this.field.addDataChangedListener(listener)
+        }
+    }
+}
+
 private class DialogExtractFunctionUi(
     private val project: Project
 ) : ExtractFunctionUi {
 
+    private fun createField(): ExtractTextField {
+        val field = NameSuggestionsField(project)
+        return /*if (field.hasSuggestions()) {
+            ExtractTextField.SuggestionTextField(field)
+        } else*/ ExtractTextField.SimpleTextField(JBTextField())
+    }
+
     override fun extract(config: RsExtractFunctionConfig, callback: () -> Unit) {
-        val functionNameField = NameSuggestionsField(project)
+        val functionNameField = this.createField()
```

Comment:
Looks like our `Extract function` doesn't support name suggestions at all.

So it can be done this way, without `ExtractTextField`:
```Kotlin
val functionNameField = NameSuggestionsField(emptyArray(), project, RsFileType)
functionNameField.minimumSize = JBUI.size(300, 30)
```
`NameSuggestionsField` creates `EditorTextField` in case of empty suggestions array. Later we can pass real suggestions instead of `emptyArray()`.